x-pack/legacy/plugins/reporting/server/usage/types.d.ts

```diff
@@ -4,38 +4,60 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-export interface AvailableTotal {
-  available: boolean;
-  total: number;
-}
-
-interface StatusCounts {
-  [statusType: string]: number;
-}
-
 export interface KeyCountBucket {
   key: string;
   doc_count: number;
 }
 
 export interface AggregationBuckets {
   buckets: KeyCountBucket[];
-  pdf?: {
-    buckets: KeyCountBucket[];
-  };
 }
 
-/*
- * Mapped Types and Intersection Types
- */
+export interface StatusByAppBucket {
+  key: string;
+  doc_count: number;
+  jobTypes: {
+    buckets: Array<{
+      doc_count: number;
+      key: string;
+      appNames: AggregationBuckets;
+    }>;
+  };
+}
 
-type AggregationKeys = 'jobTypes' | 'layoutTypes' | 'objectTypes' | 'statusTypes';
-export type AggregationResults = { [K in AggregationKeys]: AggregationBuckets } & {
+export interface AggregationResultBuckets {
```

Comment:
There are a lot of Typescript changes in this PR, because the previous `AggregationResults` type was taking a few that made expanding on it very difficult. Therefore it has been replaced with AggregationResultBuckets